### PR TITLE
Fixes #25 Npgsql 4.1 Integration Tests

### LINKED ISSUE #25: Add Integration Tests for NPGSQL 4.1 changes
Add integration tests for the changes done for the Npgsql 4.1 upgrade.

### PR DESCRIPTION
**Pull request by lxatrhea**
- Tests added for full coverage of Npgsql 4.1 changes in WebService
- Introduces CdpVersion_1_1_0 attribute for testing specific CDP4 extensions

### CODE REVIEW

File: IntegrationTestSuite/WebservicesIntegrationTests/Tests/EngineeringModel/Option/OptionTestFixture.cs [hunk 1]
```diff
@@ -18,42 +18,63 @@
...
-            var option = jArray.Single(x => (string)x[PropertyNames.Iid] == "e90e4bcd-6e17-4b75-80fb-6cded78bed57");
-            Assert.AreEqual(2, (int)option[PropertyNames.RevisionNumber]);
+            var option = jArray.Single(x => (string) x[PropertyNames.Iid] == "e90e4bcd-6e17-4b75-80fb-6cded78bed57");
+            Assert.AreEqual(2, (int) option[PropertyNames.RevisionNumber]);
             Assert.AreEqual("test option", (string) option[PropertyNames.Name]);
-            Assert.AreEqual("testoption", (string)option[PropertyNames.ShortName]);
+            Assert.AreEqual("testoption", (string) option[PropertyNames.ShortName]);
+        }
+
+        [Ignore("Reordering options is not possible in Webservice")]
```
Comment (samatrhea) [inline IntegrationTestSuite/WebservicesIntegrationTests/Tests/EngineeringModel/Option/OptionTestFixture.cs:59]: i propose we add a GH ticket reference to fix this

File: IntegrationTestSuite/WebservicesIntegrationTests/Tests/EngineeringModel/RuleVerification/RuleVerificationTestFixture.cs [hunk 2]
```diff
@@ -104,39 +104,74 @@ public void VerifyThatARuleVerificationCanBeCreatedWithWebApi()
...
+            Assert.AreEqual(2, (int) userRuleVerification[PropertyNames.RevisionNumber]);
+            Assert.AreEqual("UserRuleVerification", (string) userRuleVerification[PropertyNames.ClassKind]);
+            Assert.AreEqual("NONE", (string) userRuleVerification[PropertyNames.Status]);
+            Assert.IsNull((string) userRuleVerification[PropertyNames.ExecutedOn]);
+            Assert.IsFalse((bool) userRuleVerification[PropertyNames.IsActive]);
+            Assert.AreEqual("8a5cd66e-7313-4843-813f-37081ca81bb8", (string) userRuleVerification[PropertyNames.Rule]);
+        }
+
+        [Ignore("Doen't work because RuleVerification ordered implementation is not OK")]
+        public void VerifyThatARuleVerificationCanBeReorderedWithWebApi()
```
Comment (samatrhea) [inline IntegrationTestSuite/WebservicesIntegrationTests/Tests/EngineeringModel/RuleVerification/RuleVerificationTestFixture.cs:145]: i propose we add a GH ticket reference to fix this

File: IntegrationTestSuite/WebservicesIntegrationTests/Tests/SiteDirectory/POSTNewDomainExpertUser.json [hunk 3]
```diff
@@ -529,6 +529,28 @@
       "objectClass": "ContractChangeNotice",
       "revisionNumber": 1
     },
+    //{
```
Comment (samatrhea) [inline IntegrationTestSuite/WebservicesIntegrationTests/Tests/SiteDirectory/POSTNewDomainExpertUser.json:532]: i rather not have commented code in git, please remove comments or add a GH ticket to fix

File: IntegrationTestSuite/WebservicesIntegrationTests/Tests/EngineeringModel/Option/OptionTestFixture.cs (hunk 1 above)
Comment (lxatrhea) [reply IntegrationTestSuite/WebservicesIntegrationTests/Tests/EngineeringModel/Option/OptionTestFixture.cs:59]: Was already done. Added the issue URL to the text in the Ignore Attribute.

File: IntegrationTestSuite/WebservicesIntegrationTests/Tests/EngineeringModel/RuleVerification/RuleVerificationTestFixture.cs (hunk 2 above)
Comment (lxatrhea) [reply IntegrationTestSuite/WebservicesIntegrationTests/Tests/EngineeringModel/RuleVerification/RuleVerificationTestFixture.cs:145]: Was already done. Added the issue URL to the text in the Ignore Attribute.

**Pull request merged**